### PR TITLE
fix: #15767 unsubscribe sendgrid subuser

### LINKED ISSUE #15767: Unsubscribe people from sendgrid after deleting
we currently clean up deleted accounts manually, but it would be nice to auto-remove them from sendgrid when they delete their account from within profile: https://app.cal.com/settings/my-account/profile

CleanShot 2024-07-15 at 13 05 45@2x

this API should do it: https://www.twilio.com/docs/sendgrid/api-reference/subusers-api/delete-a-subuser

### PR DESCRIPTION
**Pull request by vijayraghav-io**
## What does this PR do?

- Fixes #15767 
- Fixes CAL-4041

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by vijayraghav-io** [comment]
@PeerRich , i needed a clarification - currently i am using cal profile's username to pass as input to `/v3/subusers/{subuser_name}` , is this ok ?
or 
should i fetch sendgrid's username
https://github.com/calcom/cal.com/blob/5becc95fc430d327d4b485246678541210fa06cd/packages/lib/Sendgrid.ts#L60
and then pass this sendgrid's username for deletion ?

**Review by anikdhabal — COMMENTED** [review]
@vijayraghav-io I think 'subuser' or 'subuser_name' is not similar to the SendGrid username. Don't look at the code, but if we create a subuser or set the subuser username using the profile 'usrname', then it is correct. Also, I read that 'New SendGrid accounts do not come with subusers; you must create them with the parent account.
[Subuser Docs](https://www.twilio.com/docs/sendgrid/ui/account-and-settings/subusers)

File: packages/lib/sync/services/SendgridService.ts [hunk 1]
```diff
@@ -83,6 +83,19 @@ export default class SendgridService extends SyncServiceCore implements ISyncSer
         return this.upsert(webUser);
       },
       delete: async (webUser: WebUserInfoType) => {
+        try {
+          const deleteSubUserResp = await this.service.sendgridRequest({
```
[the remark this answers is not in the document]
Comment (vijayraghav-io) [reply packages/lib/sync/services/SendgridService.ts]: done

**Comment by anikdhabal** [comment]
@vijayraghav-io How do you plan to test this one?

**Comment by vijayraghav-io** [comment]
This can be tested by :
1. integrating Sendgrid app with one of the cal.com account/user
   i.  Have a Sendgrid / twillio account or creds, login with this.
   ii. Generate api key for Sendgrid
   iii. Install Sendgrid with above api key in cal.com
2. Delete cal.com user/account.
    Cal.com -> Settings -> Profile -> Delete account 
3. Verify 'Step 2' also deletes SendGrid user integrated in 'Step 1'.

**Comment by vijayraghav-io** [comment]
I tried to create a twillio account couple of times, but was not possible, instead some ticket is autocreated.

**Comment by anikdhabal** [comment]
Closing this as we have doubts about whether it is really needed. I will reopen it if necessary.🙏

**Pull request closed without merging**